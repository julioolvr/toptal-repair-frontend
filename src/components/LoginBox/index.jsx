import React from 'react';
import PropTypes from 'prop-types';

class LoginBox extends React.Component {
  state = {
    username: '',
    password: '',
  };

  login(e) {
    e.preventDefault();
    this.props.onLogin(this.state.username, this.state.password);
  }

  render() {
    return (
      <form onSubmit={e => this.login(e)}>
        <div>Login</div>
        <label htmlFor="username">Username:</label>
        <input
          id="username"
          value={this.state.username}
          onChange={e => this.setState({ username: e.target.value })}
        />
        <label htmlFor="password">Password:</label>
        <input
          type="password"
          id="password"
          value={this.state.password}
          onChange={e => this.setState({ password: e.target.value })}
        />
        <button>Sign in</button>
      </form>
    );
  }
}

LoginBox.propTypes = {
  onLogin: PropTypes.func.isRequired,
};

export default LoginBox;
