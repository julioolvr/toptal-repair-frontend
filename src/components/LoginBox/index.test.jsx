import React from 'react';
import { shallow } from 'enzyme';

import LoginBox from '.';

it('renders an empty login box', () => {
  const wrapper = shallow(<LoginBox onLogin={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

it('calls the onLogin prop when submitting the form', () => {
  const onLogin = jest.fn();
  const wrapper = shallow(<LoginBox onLogin={onLogin} />);
  wrapper.find('form').simulate('submit', { preventDefault: () => {} });
  expect(onLogin).toHaveBeenCalled();
});
