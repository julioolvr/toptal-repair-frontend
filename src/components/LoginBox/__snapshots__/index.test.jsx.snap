// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an empty login box 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    Login
  </div>
  <label
    htmlFor="username"
  >
    Username:
  </label>
  <input
    id="username"
    onChange={[Function]}
    value=""
  />
  <label
    htmlFor="password"
  >
    Password:
  </label>
  <input
    id="password"
    onChange={[Function]}
    type="password"
    value=""
  />
  <button>
    Sign in
  </button>
</form>
`;
